bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/handler/BroadlinkBaseThingHandler.java

```diff
@@ -0,0 +1,285 @@
+        return true;
+    }
+
+    public void updateItemStatus() {
+        logger.trace("updateItemStatus; checking host availability at {}", thingConfig.getIpAddress());
+        if (NetworkUtils.hostAvailabilityCheck(thingConfig.getIpAddress(), 3000, logger)) {
+            logger.trace("updateItemStatus; host found at {}", thingConfig.getIpAddress());
+            if (!Utils.isOnline(getThing())) {
+                logger.trace("updateItemStatus; device not currently online, resolving");
+                transitionToOnline();
+            } else {
+                // Normal operation ...
+                boolean gotStatusOk = getStatusFromDevice();
+                if (!gotStatusOk) {
+                    if (thingConfig.isIgnoreFailedUpdates()) {
+                        logger.warn(
+                                "Problem getting status. Not marking offline because configured to ignore failed updates ...");
+                    } else {
+                        logger.error("Problem getting status. Marking as offline ...");
+                        forceOffline(ThingStatusDetail.GONE, "Problem getting status");
+                    }
+                }
+            }
+        } else {
+            if (thingConfig.isStaticIp()) {
+                if (!Utils.isOffline(getThing())) {
+                    logger.debug("Statically-IP-addressed device not found at {}", thingConfig.getIpAddress());
+                    forceOffline(ThingStatusDetail.NONE, "Couldn't find statically-IP-addressed device");
+                }
+            } else {
+                logger.debug("Dynamic IP device not found at {}, will search...", thingConfig.getIpAddress());
+                DeviceRediscoveryAgent dra = new DeviceRediscoveryAgent(thingConfig, this);
+                dra.attemptRediscovery();
+                logger.debug("Asynchronous dynamic IP device search initiated...");
+            }
+        }
+    }
+
+    public void onDeviceRediscovered(String newIpAddress) {
+        logger.info("Rediscovered this device at IP {}", newIpAddress);
+        thingConfig.setIpAddress(newIpAddress);
+        transitionToOnline();
+    }
+
+    public void onDeviceRediscoveryFailure() {
+        if (!Utils.isOffline(getThing())) {
+            logger.debug("Dynamically-IP-addressed device not found after network scan. Marking offline");
+            forceOffline(ThingStatusDetail.NONE, "Couldn't rediscover device");
+        }
+    }
+
+    private void transitionToOnline() {
+        if (!hasAuthenticated()) {
+            logger.debug(
+                    "We've never actually successfully authenticated with this device in this session. Doing so now");
+            if (authenticate()) {
+                logger.debug("Authenticated with newly-detected device, will now get its status");
+            } else {
+                logger.error("Attempting to authenticate prior to getting device status FAILED. Will mark as offline");
```

Comment:
1. Please remove any more logs to error, error should only be used if a critical error has occurred that may stop openhabs core from functioning. Warn is perhaps the level you meant however...
2. The very next line takes the thing offline and the event.log will already get the needed information logged so you dont need to log twice, in this case the logger.xxxx line should be removed completely.